### DISCUSSION TITLE
Curio SupraSeal Example hardware

### DISCUSSION THREAD

**Discussion by LexLuthr** [filecoin-project/curio#140] [General]
Welcome to our community discussion on optimal hardware configurations for running Curio SupraSeal. We encourage all community members to share their experiences, insights, and recommendations on the best hardware setups for this purpose. Whether you've found a configuration that works exceptionally well or have encountered challenges you'd like to discuss, your contributions are invaluable. Let's collaborate to build a comprehensive resource for everyone interested in maximizing the performance and efficiency of Curio SupraSeal. Share your thoughts and let's learn from each other!

The contributions must be in the below format:

## Architecture:

**Total number of pipelines:** 

**Hardware details:**

| Specification | Details |
|--------------|--------|
| **Server Name** | |
| **Function (PC1/PC2 or C2 etc)** | |
| **Number of sectors (per batch)**| |
| **CPU Model and Number** | |
| **GPU Model and Number** | |
| **Motherboard Model** | |
| **DIMM Slots** | |
| **PCIe Slots** | |
| **NVMe Over PCIe Expansion Cards? Model** | |
| **Number and Size of NVMEs (used for SupraSeal)** | |
| **[Total IOPS](https://docs.curiostorage.org/supraseal#benchmark-nvme-iops)** | |
| **Server Cost** | |
| **NVME Cost** | |
| **Time per PC1 batch** | |
| **Time per PC2 batch** | |

## What Doesn't Work?

If you've experimented with certain architectures or hardware setups that didn't perform as expected or caused issues, please share your experiences here. Your insights can help other community members avoid similar pitfalls and make more informed decisions about their setups. Let's work together to identify and learn from configurations that may not be ideal for running Curio SupraSeal effectively.

**Comment by cryptowhizzard** [comment]
**Hardware details:**

| Specification | Details |
|--------------|--------|
| **Server Name** | SupraSeal Alpha|
| **Function (PC1/PC2 or C2 etc)** | PC1 + PC2 |
| **Number of sectors (per batch)**| 64 ( 2 pipelines ) |
| **CPU Model and Number** | AMD EPYC 7H12 64-Core Processor |
| **GPU Model and Number** | Single Nvidia A5000 |
| **Motherboard Model** | HPE DL385 Gen10 |
| **DIMM Slots** | 16 |
| **PCIe Slots** | 3 |
| **NVMe Over PCIe Expansion Cards? Model** | X |
| **Number and Size of NVMEs (used for SupraSeal)** | 8 x KCD6XVUL6T40 Kioxia 6.4TB |
| **[Total IOPS](https://docs.curiostorage.org/supraseal#benchmark-nvme-iops)** | |
| **Server Cost** | |
| **NVME Cost** | |
| **Time per PC1 batch** | Between 4 and 5 Hr. |
| **Time per PC2 batch** | Between 2.5 and 3Hr |

We are in the process of building. Tuning will come in the next days.
We found out that Micron NVME's did not give us the performance we wanted thus we lost a lot of time in the setup / early test phase.

These are dual socket Chassis. We use them now as one socket , reason we can only place 8 NVME drives on the CPU. For us it was the easiest way to re-use our existing hardware and avoid high costs.

**Comment by magik6k** [comment]
## Architecture:

**Total number of pipelines:** 

**Hardware details:**

| Specification | Details |
|--------------|--------|
| **Server Name** |ASUS SERVER RS520A-E11-RS24U AMD EPYC™ 7003 2U single-socket |
| **Function (PC1/PC2 or C2 etc)** | PC1/PC2 |
| **Number of sectors (per batch)**| 32 |
| **CPU Model and Number** | CPU AMD EPYC MILAN 7313 |
| **GPU Model and Number** | RTX 3090 x1 |
| **Motherboard Model** | |
| **DIMM Slots** | 8 |
| **PCIe Slots** | 2x x16 |
| **NVMe Over PCIe Expansion Cards? Model** | |
| **Number and Size of NVMEs (used for SupraSeal)** | 10x (+2x for P2 output) Samsung PM9A3 3.84TB 2.5" U.2 NVMe G4 1DWPD 7mm SED MZQL23T8HCLS-00A07 |
| **[Total IOPS](https://docs.curiostorage.org/supraseal#benchmark-nvme-iops)** | 11100916.52 |
| **Server Cost** | ~$8700 |
| **NVME Cost** | ~$4260 |
| **Time per PC1 batch** | 4h |
| **Time per PC2 batch** | 1h25m |

EDIT 2024/11/27: NVIDIA RTX4000 Ada in this machine can complete PC2 in ~85mins, similar as a 3090

**Comment by herrehesse** [comment]
Specification | Details
-- | --
Server Name | SupraSeal Alpha
Function (PC1/PC2 or C2 etc) | PC1 + PC2
Number of sectors (per batch) | 64 ( 2 pipelines )
CPU Model and Number | AMD EPYC 7H12 64-Core Processor
GPU Model and Number | Single Nvidia A5000
Motherboard Model | HPE DL385 Gen10
DIMM Slots | 16
PCIe Slots | 3
NVMe Over PCIe Expansion Cards? Model | X
Number and Size of NVMEs (used for SupraSeal) | 8 x KCD6XVUL6T40 Kioxia 6.4TB
Total IOPS |  
Server Cost |  
NVME Cost |  
Time per PC1 batch | Between 4 and 5 Hr.
Time per PC2 batch | Between 2.5 and 3Hr

Specification	Details
Server Name	SupraSeal Ultra
Function (PC1/PC2 or C2 etc)	PC1 + PC2
Number of sectors (per batch)	64 ( 2 pipelines )
CPU Model and Number	AMD EPYC 777X 64-Core Processor
GPU Model and Number	Single Nvidia A5000
Motherboard Model	Dell R7515
DIMM Slots	16
PCIe Slots	4
NVMe Over PCIe Expansion Cards? Model	X
Number and Size of NVMEs (used for SupraSeal)	8 x KCD6XVUL6T40 Kioxia 6.4TB
[Total IOPS](https://docs.curiostorage.org/supraseal#benchmark-nvme-iops)	
Server Cost	$6.000,00 USD
NVME Cost	$6.000,00 USD
Time per PC1 batch	Between 4 and 5 Hr.
Time per PC2 batch	.5 Hr

**Reply by xiaoliwe** [reply]
@herrehesse Hidde, for the above specs of HW, What is the total daily output of the sectors? Thanks.

**Reply by cryptowhizzard** [reply]
4 to 5 batches per day of 64 sectors.

**Reply by xiaoliwe** [reply]
Thanks,and the total sectors are: 4 * 64 =256 / per day? thx.
